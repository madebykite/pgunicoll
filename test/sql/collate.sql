-- Warning: Older ICU versions (before 4.8 or 4.6) may give different results
CREATE EXTENSION pgunicoll;

SELECT * FROM (VALUES ('ÂA'), ('ÂB'), ('ÂC'), ('AA'), ('AB'), ('AC')) AS x(txt)
    ORDER BY pgunicoll(txt, '');
SELECT * FROM (VALUES ('AAA'), ('äää'), ('ÄÄÄ'), ('aaa'), ('z'), ('Z'), ('100'), ('2')) AS x(txt)
    ORDER BY pgunicoll(txt, '');
SELECT * FROM (VALUES ('z'), ('ö')) AS x(txt)
    ORDER BY pgunicoll(txt, 'de');
SELECT * FROM (VALUES ('z'), ('ö')) AS x(txt)
    ORDER BY pgunicoll(txt, 'se');
SELECT * FROM (VALUES ('of'), ('öf')) AS x(txt)
    ORDER BY pgunicoll(txt, 'de');
SELECT * FROM (VALUES ('of'), ('öf')) AS x(txt)
    ORDER BY pgunicoll(txt, 'de@collation=phonebook');
SELECT pgunicoll('foobar');
SELECT pgunicoll('ľščťžýáí');
SELECT pgunicoll('');
SELECT pgunicoll(null);
