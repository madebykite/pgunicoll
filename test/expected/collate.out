-- Warning: Older ICU versions (before 4.8 or 4.6) may give different results
CREATE EXTENSION pgunicoll;
SELECT * FROM (VALUES ('ÂA'), ('ÂB'), ('ÂC'), ('AA'), ('AB'), ('AC')) AS x(txt)
    ORDER BY pgunicoll(txt, '');
 txt 
-----
 AA
 ÂA
 AB
 ÂB
 AC
 ÂC
(6 rows)

SELECT * FROM (VALUES ('AAA'), ('äää'), ('ÄÄÄ'), ('aaa'), ('z'), ('Z'), ('100'), ('2')) AS x(txt)
    ORDER BY pgunicoll(txt, '');
 txt 
-----
 2
 100
 aaa
 AAA
 äää
 ÄÄÄ
 z
 Z
(8 rows)

SELECT * FROM (VALUES ('z'), ('ö')) AS x(txt)
    ORDER BY pgunicoll(txt, 'de');
 txt 
-----
 ö
 z
(2 rows)

SELECT * FROM (VALUES ('z'), ('ö')) AS x(txt)
    ORDER BY pgunicoll(txt, 'se');
 txt 
-----
 z
 ö
(2 rows)

SELECT * FROM (VALUES ('of'), ('öf')) AS x(txt)
    ORDER BY pgunicoll(txt, 'de');
 txt 
-----
 of
 öf
(2 rows)

SELECT * FROM (VALUES ('of'), ('öf')) AS x(txt)
    ORDER BY pgunicoll(txt, 'de@collation=phonebook');
 txt 
-----
 öf
 of
(2 rows)

SELECT pgunicoll('foobar');
        pgunicoll         
--------------------------
 \x3345452b294b010a010a00
(1 row)

SELECT pgunicoll('ľščťžýáí');
                         pgunicoll                          
------------------------------------------------------------
 \x3f4d2d4f5b5929390145904590459045904590458845884588011400
(1 row)

SELECT pgunicoll('');
 pgunicoll 
-----------
 \x010100
(1 row)

SELECT pgunicoll(null);
 pgunicoll 
-----------
 
(1 row)

